src/perf_counters.h

```diff
@@ -124,44 +126,44 @@ class PerfCounters final {
 // Typical usage of the above primitives.
 class PerfCountersMeasurement final {
  public:
-  PerfCountersMeasurement(PerfCounters&& c)
-      : counters_(std::move(c)),
-        start_values_(counters_.IsValid() ? counters_.names().size() : 0),
-        end_values_(counters_.IsValid() ? counters_.names().size() : 0) {}
+  PerfCountersMeasurement(const std::vector<std::string>& counter_names);
+  ~PerfCountersMeasurement();
 
-  bool IsValid() const { return counters_.IsValid(); }
+  bool IsValid() const { return counters_->IsValid(); }
 
   BENCHMARK_ALWAYS_INLINE void Start() {
     assert(IsValid());
     // Tell the compiler to not move instructions above/below where we take
     // the snapshot.
     ClobberMemory();
-    counters_.Snapshot(&start_values_);
+    valid_read_ &= counters_->Snapshot(&start_values_);
     ClobberMemory();
   }
 
-  BENCHMARK_ALWAYS_INLINE std::vector<std::pair<std::string, double>>
-  StopAndGetMeasurements() {
+  BENCHMARK_ALWAYS_INLINE bool Stop(std::vector<std::pair<std::string, double>> *measurements) {
```

Comment:
Why did we rename this to `Stop`?

Reply:
I did it to keep consistance, 
because when I moved the result measurements from return value to the parameter, 
I noticed there are other `void ::Stop(Result* result)`.